src/lightning_lite/plugins/precision/precision.py

```diff
@@ -34,6 +36,20 @@ def forward_context(self) -> Generator[None, None, None]:
         """A contextmanager for managing model forward/training_step/evaluation_step/predict_step."""
         yield
 
+    def convert_input(self, data: Tensor) -> Tensor:
+        """Convert model inputs (forward) to the floating point precision type of this plugin.
+
+        This is a no-op for tensors that are not of floating-point type or already have the desired type.
+        """
+        return _convert_fp_tensor(data, torch.float32)
```

Comment:
Couldn't we do this in the base

```suggestion
        precision_to_type = {"bf16": torch.bfloat16, 16: torch.float16, 32: torch.float32}
        dst_type = precision_to_type[self.precision]
        return _convert_fp_tensor(data, dst_type)
```

so that none of the subclasses need to override it?

Reply:
Ideally, I want to mark theses interfaces abstract to force all of them to implement it. But I can't as this would bring more breaking changes on the PL side. 

Since some plugins can only manage a subset of precision types, I think it would be wrong if the super implementation could handle more than allowed.

Reply:
I prefer having that as abstract interface.

Reply:
Some options:

1. Current
2. Add abstract now: Small breaking change for custom plugins (who makes them?). Plus, when #14798 merges, we need to provide the implementations for PL plugins
3. Add `TODO(lite): make abstract` to delay it

Reply:
If the PL classes will not define it. It's boilerplate-y to mark it as abstract